### PR TITLE
[TACHYON-219] Removed MasterClient.shutdown() and moved its functionality in the close...

### PR DESCRIPTION
**Pull request by abhirajbutala**
...() call. Renamed the previous close() to disconnect(), as it allows the user to temporarily disconnect with the Tachyon Master. Appropriate updates to the unit tests. Minor edits to the comments and log statements.

### CODE REVIEW

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
@haoyuan can you schedule a test for this?

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by abhirajbutala** [comment]
That's strange, the tests pass locally for me (tried again right now):

;
;
Running tachyon.master.BlockInfoTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 sec - in tachyon.master.BlockInfoTest
Running tachyon.master.MasterClientTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 8.637 sec - in tachyon.master.MasterClientTest
Running tachyon.command.TFsShellTest
Tests run: 36, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 116.071 sec - in tachyon.command.TFsShellTest
;
;
;


[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Tachyon Under File System ......................... SUCCESS [17.068s]
[INFO] Tachyon Under File System - HDFS .................. SUCCESS [15:11.468s]
[INFO] Tachyon Under File System - Gluster FS ............ SUCCESS [10.267s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15:39.775s
[INFO] Finished at: Wed Apr 15 16:58:39 PDT 2015
[INFO] Final Memory: 30M/73M
[INFO] ------------------------------------------------------------------------

**Comment by calvinjia** [comment]
Jenkins, retest this please.

**Comment by calvinjia** [comment]
@abhirajbutala Yeah shouldn't be your issue, this test is known to be flakey. See https://tachyon.atlassian.net/browse/TACHYON-321

File: core/src/test/java/tachyon/master/MasterClientTest.java [hunk 1]
```diff
@@ -64,11 +64,12 @@ public void openCloseTest() throws FileAlreadyExistException, InvalidPathExcepti
     Assert.assertTrue(masterClient.isConnected());
     masterClient.user_createFile("/file", "", Constants.DEFAULT_BLOCK_SIZE_BYTE, true);
     Assert.assertTrue(masterClient.getFileStatus(-1, "/file") != null);
-    masterClient.close();
+    masterClient.disconnect();
     Assert.assertFalse(masterClient.isConnected());
     masterClient.connect();
     Assert.assertTrue(masterClient.isConnected());
     Assert.assertTrue(masterClient.getFileStatus(-1, "/file") != null);
+    masterClient.close();
```
Comment (calvinjia) [inline core/src/test/java/tachyon/master/MasterClientTest.java]: I don't think you need to call close here.

**Comment by calvinjia** [comment]
@abhirajbutala Could you verify that you replaced all the usages of close/shutdown with the new methods? I think one place that wasn't changed is in `MasterClientHeartbeatExecutor`?

**Comment by abhirajbutala** [comment]
Thanks for reviewing @calvinjia, and good catch. Have addressed your comments and verified that all usages are replaced with new methods.

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
+1 LGTM, the other `close()` methods were not changed to `disconnect()` in the unit tests but I think `close()` is more appropriate there anyway

**Comment by abhirajbutala** [comment]
Yes, purposefully kept them as close() because there aren't any operations after that in those tests. Thanks @calvinjia.

**Pull request merged**